internal/orchestrator/cache.go

```diff
@@ -0,0 +1,39 @@
+package orchestrator
+
+import (
+	"context"
+	"errors"
+
+	"github.com/docker/cli/cli/command"
+
+	"github.com/arduino/arduino-app-cli/internal/orchestrator/app"
+)
+
+type CleanAppCacheRequest struct {
+	ForceClean bool
+}
+
+var ErrCleanCacheRunningApp = errors.New("cannot remove cache of a running app")
+
+// CleanAppCache removes the `.cache` folder. If it detects that the app is running
+// it tries to stop it first.
+func CleanAppCache(
+	ctx context.Context,
+	docker command.Cli,
+	app app.ArduinoApp,
+	req CleanAppCacheRequest,
+) error {
+	runningApp, err := getRunningApp(ctx, docker.Client())
+	if err != nil {
+		return err
+	}
+	if runningApp.FullPath.EqualsTo(app.FullPath) {
+		if !req.ForceClean {
+			return ErrCleanCacheRunningApp
+		}
+		// We try to remove docker related resources at best effort
+		_ = StopAndDestroyApp(ctx, app)
```

Comment:
This is an iterator, so you should consume it, otherwise it will not execute anything

```suggestion
		for range StopAndDestroyApp(ctx, app) {
			// just consume the iterator
		}
```